v2/parser/parse.go

```diff
@@ -143,6 +154,14 @@ func (p *Parser) LoadPackages(patterns ...string) error {
 	return err
 }
 
+// LoadPackagesWithConfig loads and parses the specified Go packages with the
+// specified packages.Config as a starting point.  This is for testing, and
+// only the .Dir and .Env fields of the Config will be considered.
```

Comment:
can you add a guard that only Dir and Env and error for now if anything else is set, to keep from accumulating usage that sets other fields and may break in the future if we decide to honor additional fields?

Reply:
Unfortunately go's `packagestest` sets other fields, which would break this check:

```
        Config: &packages.Config{    
            Dir:     temp,    
            Env:     append(os.Environ(), "GOPACKAGESDRIVER=off", "GOROOT="), // Clear GOROOT to work around #32849.     
            Overlay: make(map[string][]byte),    
            Tests:   true,    
            Mode:    packages.LoadImports,    
        }, 
```

So a) it's good fortune that I was already setting `Tests: false`; b) it doesn't seem to need all of LoadImports, and works with our own `Mode`.

If you want to be super-paranoid, I can take `dir string, env []string` instead of `cfg *packages.Config` in the public API, and convert that to a smaller struct internally.  It just feels like a lot of work :)

Or I can rename the public function like `LoadPackagesWithConfigForTesting` and rely on the fact that this is not REALLY a public project anymore.

Reply:
eh, that's probably good enough

/lgtm
/approve